backend_new/src/controllers/ami-chart.controller.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+  Query,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import {
+  ApiExtraModels,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags,
+} from '@nestjs/swagger';
+import { AmiChartService } from '../services/ami-chart.service';
+import { AmiChart } from '../dtos/ami-charts/ami-chart-get.dto';
+import { AmiChartCreate } from '../dtos/ami-charts/ami-chart-create.dto';
+import { AmiChartUpdate } from '../dtos/ami-charts/ami-chart-update.dto';
+import { defaultValidationPipeOptions } from '../utilities/default-validation-pipe-options';
+import { AmiChartQueryParams } from '../dtos/ami-charts/ami-chart-query-params.dto';
+import { IdDTO } from '../dtos/shared/id.dto';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+
+@Controller('/amiCharts')
+@ApiTags('amiCharts')
+@UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+@ApiExtraModels(AmiChartCreate, AmiChartUpdate, IdDTO, AmiChartQueryParams)
+export class AmiChartController {
+  constructor(private readonly AmiChartService: AmiChartService) {}
+
+  @Get()
+  @ApiOperation({ summary: 'List amiCharts', operationId: 'list' })
+  @ApiOkResponse({ type: AmiChart, isArray: true })
+  async list(@Query() queryParams: AmiChartQueryParams) {
```

Comment:
suggestion: let's be thorough and specify the return type of functions.

Reply:
function return types should now be specified!